lms/static/scripts/frontend_apps/components/LMSFilePicker.tsx

```diff
@@ -184,18 +184,53 @@ export default function LMSFilePicker({
     }
   };
 
+  // Does the actual file fetching for the active directory, either by loading
+  // it from its children, or by going to the API to get them
+  const loadFilesToDisplay = useCallback(
+    (isReload = false): Promise<File[]> => {
+      const getNextAPICallInfo = () =>
+        folderPath[folderPath.length - 1]?.contents || listFilesApi;
+      const loadFilesFromApi = (): Promise<File[]> =>
+        apiCall({
+          authToken,
+          path: getNextAPICallInfo().path,
+        });
+
+      if (!isReload) {
+        // If the last element in current folderPath contains children, we use
+        // them without calling the API. Otherwise, we fetch files from the API
+        const children = folderPath[folderPath.length - 1]?.children;
+        return children ? Promise.resolve(children) : loadFilesFromApi();
+      }
+
+      return loadFilesFromApi().then(loadedFiles => {
```

Comment:
Oh! I should have called this out in the last round, but since we did the review verbally I missed it. Our convention is not to mix `Promise` chaining with `async/await` code where possible (we have a general preference for `async` over `Promises` when feasible).

Could convert this to an `await` (you'll have to make a couple of functions `aync` to do it)?

Reply:
I agree with that approach. I also prefer to use `async`/`await` when possible.

In this case I ended up using promises because the first version of this function looked more or less like this:

```ts
const loadFiles = async (): Promise<File[]> => {
  if (!isReload) {
    return folderPath[folderPath.length - 1]?.children ?? loadFilesFromApi();
  }

  return loadFilesFromApi();
};
```

However, this made some tests fail, and I had to end up with this:

```ts
if (!isReload) {
  return folderPath[folderPath.length - 1]?.children ?? (await loadFilesFromApi());
}

// ...
```

Doing `return await ...` always feels awkward. It seems like something you shouldn't need, but for some reason it didn't work.

I didn't dig too much on it though. I'll give it another thought. There has to be a way.

But ultimately, I ended up making `loadFiles` not `async`, and as a side effect, I also had to do `loadFilesFromApi().then(...)` when it's a reload, instead of `await loadFilesFromApi(); ...`, which relates with this other comment: https://github.com/hypothesis/lms/pull/4853#discussion_r1081659272